### PR TITLE
feat: build rpc connection

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Closes MF-0000

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/base/src/specs/index.ts [hunk 1]
```diff
@@ -884,10 +884,15 @@ export interface AddressBookState extends Startable {
 }
 
 export interface NetworkState<ChainId, SchemaType, NetworkType> extends Startable {
+    /** The id of the used network to build connection. */
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    /** The id of the used network. */
```

File: packages/web3-shared/base/src/specs/index.ts [hunk 2]
```diff
@@ -884,10 +884,15 @@ export interface AddressBookState extends Startable {
...
 
 export interface NetworkState<ChainId, SchemaType, NetworkType> extends Startable {
+    /** The id of the used network. */
+    networkID?: Subscription<string>
+    /** All available networks. */
     networks?: Subscription<Array<ReasonableNetwork<ChainId, SchemaType, NetworkType>>>
 
     /** Add a new network. */
     addNetwork: (descriptor: TransferableNetwork<ChainId, SchemaType, NetworkType>) => Promise<void>
+    /** Use the network RPC and build connection. */
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    /** Use the network RPC to build a connection. */
```

File: packages/web3-providers/src/Web3/Base/state/Network.ts [hunk 3]
```diff
@@ -95,6 +130,9 @@ export class NetworkState<ChainId, SchemaType, NetworkType>
     async removeNetwork(id: string) {
         this.assertNetwork(id)
 
-        await this.storage.setValue(omit(this.storage.value, id))
+        await Promise.all([
+            this.storage.networks.setValue(omit(this.storage.networks.value, id)),
+            this.networkID?.getCurrentValue() === id ? await this.storage.networkID.setValue('') : Promise.resolve(),
```
Comment (guanbinrui) [inline packages/web3-providers/src/Web3/Base/state/Network.ts:135]:
```suggestion
            this.networkID?.getCurrentValue() === id ? this.storage.networkID.setValue('') : Promise.resolve(),
```

File: packages/web3-providers/src/Web3/Base/state/AddressBook.ts [hunk 4]
```diff
@@ -10,12 +10,13 @@ export class AddressBookState implements Web3AddressBookState {
     constructor(
         protected context: Plugin.Shared.SharedUIContext,
         protected options: {
+            pluginID: NetworkPluginID
             isValidAddress(a: string): boolean
             isSameAddress(a: string, b: string): boolean
             formatAddress(a: string): string
         },
     ) {
-        const { storage } = this.context.createKVStorage('persistent', {}).createSubScope('AddressBookV2', {
```
Comment (Jack-Works) [inline packages/web3-providers/src/Web3/Base/state/AddressBook.ts:18]: what about old data?

File: packages/web3-providers/src/Web3/Base/state/Transaction.ts [hunk 5]
```diff
@@ -33,11 +39,12 @@ export class TransactionState<ChainId extends PropertyKey, Transaction>
             chainId?: Subscription<ChainId>
         },
         protected options: {
+            pluginID: NetworkPluginID
             formatAddress(a: string): string
             isValidChainId(chainId?: ChainId): boolean
         },
     ) {
-        const { storage } = this.context.createKVStorage('persistent', {}).createSubScope('Transaction', {
```
Comment (Jack-Works) [inline packages/web3-providers/src/Web3/Base/state/Transaction.ts:40]: what about old data?

File: packages/web3-providers/src/Web3/Base/state/Token.ts [hunk 6]
```diff
@@ -82,12 +89,13 @@ export class TokenState<ChainId extends number, SchemaType> implements Web3Token
             chainId?: Subscription<ChainId>
         },
         protected options: {
+            pluginID: NetworkPluginID
             isValidAddress(a: string): boolean
             isSameAddress(a: string, b: string): boolean
             formatAddress(a: string): string
         },
     ) {
-        const { storage } = context.createKVStorage('persistent', {}).createSubScope('Token', defaultValue)
```
Comment (Jack-Works) [inline packages/web3-providers/src/Web3/Base/state/Token.ts:90]: what about old data?

**Pull request merged**